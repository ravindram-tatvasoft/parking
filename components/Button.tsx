import {StyleSheet, TouchableOpacity, Text, TouchableOpacityProps, ViewStyle } from "react-native"
import { colors } from "../utils"

export interface ButtonProps extends TouchableOpacityProps {
    name: string,
    onPressButton: () => void,
    testId: string
    rootStyle?: ViewStyle | ViewStyle[]
}

export const Button = (props: ButtonProps) => {
    const { name, onPressButton, testId, rootStyle, ...rest} = props
    return(<TouchableOpacity testID={testId} onPress={onPressButton} style={[styles.root, rootStyle || []]}>
        <Text style={styles.txtName}>{name}</Text>
    </TouchableOpacity>)
}


const styles = StyleSheet.create({
    root: {
        backgroundColor:colors.primary,
        paddingHorizontal:20,
        paddingVertical: 10
    },
    txtName: {
        color: 'white',
        textAlign: "center"
    }
})