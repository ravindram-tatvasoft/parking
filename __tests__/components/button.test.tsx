// __tests__
import { render } from "@testing-library/react-native";
import * as React from "react";
import { Button, ButtonProps } from "../../components";


describe('Component: Button ', ()=> {
    test(`renders correctly`, () => {
        const props: ButtonProps = {
            name: "Button",
            testId: "btnId",
            onPressButton: jest.fn()
        }
        const tree = render(<Button {...props} />);
        expect(tree).toMatchSnapshot();
    });
})
