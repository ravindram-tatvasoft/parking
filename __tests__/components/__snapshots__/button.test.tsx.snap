// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button  renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#246EE9",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
  testID="btnId"
>
  <Text
    style={
      Object {
        "color": "white",
        "textAlign": "center",
      }
    }
  >
    Button
  </Text>
</View>
`;
